lib/components/mixins/resizable.dart

```diff
@@ -1,21 +1,19 @@
+import 'package:flame/components/component.dart';
+import 'package:meta/meta.dart';
+
 import '../../extensions/vector2.dart';
 
-/// Useful mixin to add to your components if you want to hold a reference to the current screen size.
-///
-/// This mixin implements the resize method in order to hold an updated reference to the current screen [size].
-/// Also, it updates its [children], if any.
-class Resizable {
+/// A [Component] mixin to make your component keep track of the size opf the game viewport.
+mixin Resizable on Component {
   /// This is the current updated screen size.
-  Vector2 size;
+  Vector2 gameSize;
 
   /// Implementation provided by this mixin to the resize hook.
-  void resize(Vector2 size) {
-    this.size = size;
-    resizableChildren().forEach((e) => e?.resize(size));
+  /// This is a hook called by [BaseGame] to let this component know that the screen (or flame draw area) has been update.
+  @override
+  @mustCallSuper
+  void onGameResize(Vector2 gameSize) {
+    super.onGameResize(gameSize);
+    this.gameSize = gameSize;
   }
-
-  /// Overwrite this to add children to this [Resizable].
-  ///
-  /// If a [Resizable] has children, its children as resized as well.
-  Iterable<Resizable> resizableChildren() => [];
```

Comment:
How will the resize events propagate to children in the new flow?

Reply:
Position component does that already.